lib/util/git/modified-cache.ts

```diff
@@ -0,0 +1,41 @@
+import is from '@sindresorhus/is';
+import { getCache } from '../cache/repository';
+import type { BranchCache } from '../cache/repository/types';
+
+export function getCachedModifiedResult(
+  branchName: string,
+  branchSha: string
+): boolean | null {
+  const { branches } = getCache();
+  const branch = branches?.find((branch) => branch.branchName === branchName);
+
+  if (branch?.sha !== branchSha) {
+    return null;
+  }
+
+  return branch.isModified;
+}
+
+export function setCachedModifiedResult(
+  branchName: string,
+  branchSha: string,
+  isModified: boolean
+): void {
+  const cache = getCache();
+  cache.branches ??= [];
+  const { branches } = cache;
+  const branch =
+    branches?.find((branch) => branch.branchName === branchName) ??
+    ({ branchName: branchName } as BranchCache);
```

Comment:
do we really need the cast?

Reply:
Yes. We will be using properties present in BranchCache like `sha` and `isModified` so its necessary for them to be present in the object type.